api/src/main/java/org/apache/iceberg/ManifestFile.java

```diff
@@ -199,6 +200,16 @@ default boolean hasDeletedFiles() {
      */
     boolean containsNull();
 
+    /**
+     * Returns true if at least one data file in the manifest has a nan value for the field.
+     * Null if this information doesn't exist.
+     * <p>
+     * Default to return null to ensure backward compatibility.
+     */
+    default Boolean containsNaN() {
+      return null;
```

Comment:
I think `PartitionSummary` is populated when trying to write a file, so if the code contains this change, it will populate and write NaN boolean correctly. I guess you might be thinking about `GenericPartitionFieldSummary` when you mention returning true as default since that's the class to be constructed when reading from avro files that may not contain this info?

Reply:
I see, make sense, thanks for the explanation.